src/components/Calendar.ts

```diff
@@ -0,0 +1,146 @@
+import { Component, ReactChild, createElement } from "react";
+
+import { Alert } from "./Alert";
+import * as BigCalendar from "react-big-calendar";
+import { CalendarLoading } from "./CalendarLoading";
+import { DragDropContext } from "react-dnd";
+import HTML5Backend from "react-dnd-html5-backend";
+import * as globalize from "globalize";
+import localizer from "react-big-calendar/lib/localizers/globalize";
+import * as withDragAndDrop from "react-big-calendar/lib/addons/dragAndDrop";
+
+import "react-big-calendar/lib/addons/dragAndDrop/styles.css";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import "../ui/Calendar.scss";
+
+localizer(globalize);
+const DragAndDropCalendar = withDragAndDrop(BigCalendar);
+
+export interface CalendarProps {
+    alertMessage?: ReactChild;
+    events?: CalendarEvent[];
+    color: string;
+    loading?: boolean;
+    showMultiDayTimes?: boolean;
+    defaultView: View;
+    startPosition: Date;
+    firstDay?: number;
+    popup: boolean;
+    selectable: boolean;
+    dayFormat?: string;
+    weekdayFormat?: string;
+    timeGutterFormat?: string;
+    monthHeaderFormat?: string;
+    dayHeaderFormat?: string;
+    onSelectEventAction?: (eventInfo: object) => void;
+    onSelectSlotAction?: (slotInfo: object) => void;
+    onEventDropAction?: (eventInfo: object) => void;
+}
+
+export type View = "month" | "week" | "work_week" | "day" | "agenda";
```

Comment:
Stick to camel case workWeek

Reply:
`work_week` is a key work combination from the library